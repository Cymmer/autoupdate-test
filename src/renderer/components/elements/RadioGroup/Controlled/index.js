import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import styles from '../radio.module.scss';

import Icon from '../../Icon';

import Label from '../../Label';

import radioTypes from '../constants/radioTypes';

const ControlledRadio = ({
  name,
  label,
  id,
  checked,
  className,
  onChange,
  disabled,
  value,
  type,
  icon,
}) => (
  <div
    className={cn(className, styles[`Radio___${type}`], {
      [styles[`Radio___${type}___disabled`]]: disabled,
    })}
    id={id ? `${id}Container` : null}
  >
    <input
      checked={checked}
      className={styles[`Radio___${type}_input`]}
      data-test="radio"
      disabled={disabled}
      id={id}
      name={name}
      type="radio"
      value={value}
      onChange={onChange}
    />
    <Label
      className={styles[`Radio___${type}_label`]}
      disabled={disabled}
      htmlFor={id}
    >
      {icon && type === radioTypes.LARGE && (
        <Icon className={styles.Radio___large_icon} icon={icon} />
      )}
      {label}
    </Label>
  </div>
);

ControlledRadio.defaultProps = {
  checked: false,
  className: null,
  disabled: false,
  type: radioTypes.SMALL,
  icon: null,
};

ControlledRadio.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  checked: PropTypes.bool,
  id: PropTypes.string.isRequired,
  className: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.any.isRequired,
  disabled: PropTypes.bool,
  type: PropTypes.oneOf([radioTypes.SMALL, radioTypes.LARGE]),
  icon: PropTypes.string,
};

export default ControlledRadio;
