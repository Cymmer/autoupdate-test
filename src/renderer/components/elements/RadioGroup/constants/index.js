export { default as radioTypes } from './radioTypes';
