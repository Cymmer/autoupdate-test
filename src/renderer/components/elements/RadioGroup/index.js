import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import GLOBALS from 'codechum-app-globals';
import styles from './styles.module.scss';

import Tooltip from '../Tooltip';
import tooltipPlacement from '../Tooltip/constants/tooltipPlacement';
import Text from '../Text';
import Span from '../Text/Span';
import textTypes from '../Text/constants/textTypes';

import ControlledRadio from './Controlled';

import radioTypes from './constants/radioTypes';
import RequiredField from '../RequiredField';

const RadioGroup = ({
  label,
  isRequired,
  className,
  innerClassName,
  options,
  checked,
  disabled,
  type,
  error,
  hasNewBadge,
}) => (
  <div className={cn(styles.RadioGroup_container, className)}>
    {label && (
      <Text
        className={styles.RadioGroup_label}
        colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['400']}
        type={textTypes.BODY.SM}
      >
        {isRequired ? <RequiredField placeholder={label} /> : label}
        {hasNewBadge && (
          <Span className={styles.RadioGroup_newBadge} type={textTypes.DATA.XS}>
            NEW!
          </Span>
        )}
      </Text>
    )}
    <div className={cn(styles.RadioGroup, innerClassName)}>
      {options?.map(
        ({ id, value, label: radioLabel, name, onChange, icon, tooltip }) =>
          tooltip ? (
            <Tooltip content={tooltip} placement={tooltipPlacement.BOTTOM}>
              <ControlledRadio
                key={id}
                checked={checked === value}
                className={styles[`RadioGroup___${type}_radio`]}
                disabled={disabled}
                icon={icon}
                id={id}
                label={radioLabel}
                name={name}
                type={type}
                value={value}
                onChange={onChange}
              />
            </Tooltip>
          ) : (
            <ControlledRadio
              key={id}
              checked={checked === value}
              className={styles[`RadioGroup___${type}_radio`]}
              disabled={disabled}
              icon={icon}
              id={id}
              label={radioLabel}
              name={name}
              type={type}
              value={value}
              onChange={onChange}
            />
          )
      )}
    </div>
    {error && (
      <div className={styles.RadioGroup_helperTextContainer}>
        <Text
          colorClass={GLOBALS.COLOR_CLASSES.RED['500']}
          type={textTypes.BODY.XS}
        >
          {error}
        </Text>
      </div>
    )}
  </div>
);

RadioGroup.defaultProps = {
  className: null,
  innerClassName: null,
  label: null,
  error: null,
  disabled: false,
  type: radioTypes.SMALL,
  checked: null,
  hasNewBadge: false,
};

RadioGroup.propTypes = {
  className: PropTypes.string,
  innerClassName: PropTypes.string,
  label: PropTypes.string,
  error: PropTypes.string,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      value: PropTypes.string,
      label: PropTypes.string,
      name: PropTypes.string,
      onChange: PropTypes.func,
      icon: PropTypes.string,
    })
  ).isRequired,
  checked: PropTypes.string,
  disabled: PropTypes.bool,
  type: PropTypes.oneOf([radioTypes.SMALL, radioTypes.LARGE]),
  hasNewBadge: PropTypes.bool,
  isRequired: PropTypes.bool,
};

export default RadioGroup;
