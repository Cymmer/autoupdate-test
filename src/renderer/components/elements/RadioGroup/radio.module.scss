@import '../../../scss/variables';
@import '../../../scss/mixins';
@import '../../../scss/themes';

.Radio {
  &___small {
    &___disabled {
      opacity: 0.5;
    }

    &_label {
      display: flex;
      position: relative;
      padding-left: 26px;
      user-select: none;
      font-size: 1rem;

      &::before {
        content: '';
        display: block;
        width: 16px;
        height: 16px;
        margin: -1px 0 0 0;
        position: absolute;
        top: 1px;
        left: 0;
        border: 2px solid;
        @include theme {
          border-color: t('textColor-400');
        }
        background-color: transparent;
        border-radius: 50%;
      }

      &:hover {
        &::before {
          border-color: $blue-300 !important;
        }
      }
    }

    &_input {
      display: none;

      &:checked + .Radio___small_label::after {
        content: '';
        display: block;
        position: absolute;
        top: 5px;
        left: 5px;
        width: 10px;
        height: 10px;
        border-radius: 50%;
        background: $blue-300;
      }

      &:checked + .Radio___small_label::before {
        border-color: $blue-300;
      }
    }
  }

  &___large {
    @include flex-center;

    &___disabled {
      opacity: 0.5;
    }

    &_label {
      @include ease(0.2s);
      @include flex-center;
      flex-direction: column;
      @include heading(1rem, 1.25rem);
      margin: 8px 0 0 0;
      width: 180px;
      height: 192px;
      border-radius: 16px;
      border: 1px solid;
      @include theme {
        border-color: t('borderColor');
        color: t('textColor-400');
      }

      &:hover {
        color: $blue-300 !important;
      }
    }

    &_icon {
      font-size: 6rem;
      color: inherit;
    }

    &_input {
      display: none;

      &:checked + .Radio___large_label {
        background-color: transparentize($blue-300, 0.9);
        color: $blue-300;
        border: 1px solid $blue-300;
      }
    }
  }
}
