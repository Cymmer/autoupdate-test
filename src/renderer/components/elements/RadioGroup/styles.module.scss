@use '../../../scss' as *;

.RadioGroup {
  display: flex;
  flex-wrap: wrap;

  &_label {
    margin: 0 0 6px 0;
  }

  &_container {
    display: flex;
    flex-direction: column;
  }

  &___small {
    &_radio {
      margin: 8px 16px 8px 0;
    }
  }

  &___large {
    &_radio {
      margin: 0 32px 0 0;

      &:last-child {
        margin-right: 0;
      }
    }
  }

  &_helperTextContainer {
    margin: 8px 0 0 16px;
  }

  &_newBadge {
    margin: 0 0 0 8px;
    padding: 0.25em 0.5em;
    color: $neutral-0;
    background: $bg-gradient;
    border-radius: 8px;
  }
}
