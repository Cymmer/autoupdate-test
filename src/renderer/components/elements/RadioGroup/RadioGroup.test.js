import React from 'react';
import { shallow } from 'enzyme';
import RadioGroup from './index';

describe('RENDERING - RadioGroup', () => {
  it('should render with default props', () => {
    const component = shallow(<RadioGroup options={[]} />);
    expect(component).toMatchSnapshot();
  });

  it('should render a label if it has a label', () => {
    const component = shallow(<RadioGroup label="Test" options={[]} />);
    expect(component).toMatchSnapshot();
  });

  it('should render the appropriate number of options', () => {
    const component = shallow(
      <RadioGroup
        label="Test"
        options={[
          {
            id: 'test1',
            value: 'test1',
            label: 'Test 1',
            name: 'testName',
            onChange: () => {},
            icon: 'test1',
          },
          {
            id: 'test2',
            value: 'test2',
            label: 'Test 2',
            name: 'testName',
            onChange: () => {},
            icon: 'test2',
          },
        ]}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render an error if it has an error', () => {
    const component = shallow(<RadioGroup error="Test" options={[]} />);
    expect(component).toMatchSnapshot();
  });
});
