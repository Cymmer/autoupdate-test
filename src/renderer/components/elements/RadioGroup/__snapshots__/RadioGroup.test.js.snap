// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - RadioGroup should render a label if it has a label 1`] = `
<div
  className="RadioGroup_container"
>
  <Text
    className="RadioGroup_label"
    colorClass="n___400"
    element={null}
    id={null}
    type="body___sm"
  >
    Test
  </Text>
  <div
    className="RadioGroup"
  />
</div>
`;

exports[`RENDERING - RadioGroup should render an error if it has an error 1`] = `
<div
  className="RadioGroup_container"
>
  <div
    className="RadioGroup"
  />
  <div
    className="RadioGroup_helperTextContainer"
  >
    <Text
      className={null}
      colorClass="r___500"
      element={null}
      id={null}
      type="body___xs"
    >
      Test
    </Text>
  </div>
</div>
`;

exports[`RENDERING - RadioGroup should render the appropriate number of options 1`] = `
<div
  className="RadioGroup_container"
>
  <Text
    className="RadioGroup_label"
    colorClass="n___400"
    element={null}
    id={null}
    type="body___sm"
  >
    Test
  </Text>
  <div
    className="RadioGroup"
  >
    <ControlledRadio
      checked={false}
      className="RadioGroup___small_radio"
      disabled={false}
      icon="test1"
      id="test1"
      key="test1"
      label="Test 1"
      name="testName"
      onChange={[Function]}
      type="small"
      value="test1"
    />
    <ControlledRadio
      checked={false}
      className="RadioGroup___small_radio"
      disabled={false}
      icon="test2"
      id="test2"
      key="test2"
      label="Test 2"
      name="testName"
      onChange={[Function]}
      type="small"
      value="test2"
    />
  </div>
</div>
`;

exports[`RENDERING - RadioGroup should render with default props 1`] = `
<div
  className="RadioGroup_container"
>
  <div
    className="RadioGroup"
  />
</div>
`;
